sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -83,13 +78,21 @@ export class DefaultManagedIdentityCredential extends ManagedIdentityCredential
       ?.managedIdentityResourceId;
 
     // ManagedIdentityCredential throws if both the resourceId and the clientId are provided.
-    const managedIdentityOptions: ManagedIdentityCredentialOptions = {
-      resourceId: managedResourceId,
-      clientId: managedIdentityClientId,
-      ...options,
-    };
-
-    super(managedIdentityOptions);
+    // let managedIdentityOptions;
+    if (managedResourceId) {
+      const managedIdentityResourceIdOptions: ManagedIdentityCredentialResourceIdOptions = {
+        ...options,
+      };
+      managedIdentityResourceIdOptions.resourceId = managedResourceId;
```

Comment:
```suggestion
      const managedIdentityResourceIdOptions: ManagedIdentityCredentialResourceIdOptions = {
        ...options,
        resourceId: managedResourceId,
      };
```

nit: same for the other one below this

Reply:
updated